File: AEDS-II/verde/Palindromos/palindromoC/palindromoRec.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool palindromoRec(char* string, int index, bool palindromo) {
    
    int tam = strlen(string);

    if (index < tam / 2) {
        if(string[index] != string[tam - 1 - index]){
     
            palindromo = false;
        }   
    }
    else if (index >= tam) {
        return palindromo;
    }

    return palindromoRec(string, index + 1, palindromo);
}

int main() {

    char string[1000];
    scanf(" %[^\n\r]", string);

    bool palindromo = true;
    
    while (strcmp(string, "FIM")) {
        
        if (palindromoRec(string, 0, palindromo)) {
            printf("SIM\n");       
        } else {
            printf("NÃO\n");
        }
        
        scanf(" %[^\n\r]", string);
    }
    
    return 0;
}

File: AEDS-II/verde/Palindromos/PalindromoJAVA/PalindromoRec/PalindromoRec.java
import java.util.Scanner;

public class PalindromoRec {

    public static boolean palindromo(String string, int tam, int index) {

        if(index >= tam) {
            return true;
        }

        if(string.charAt(index) != string.charAt(tam)) {
            return false;
        }
        return palindromo(string, tam - 1, index + 1);
    }

    public static void main(String[] args) {
        
        Scanner scanner = new Scanner(System.in);
        String entrada = scanner.nextLine();
        

        while(!(entrada.length() == 3 && entrada.charAt(0) == 'F' && entrada.charAt(1) == 'I' && entrada.charAt(2) == 'M')){ 

            
            int tam = entrada.length() - 1;
            if(palindromo(entrada, tam, 0)){
                
                System.out.println("SIM");
            }
            else{
                System.out.println("NAO");
            }

            entrada = scanner.nextLine();
        }

        scanner.close();
    }
}
